arch/arm/core/aarch64/vector_table.S

```diff
@@ -13,10 +13,50 @@
 #include <arch/cpu.h>
 #include <offsets_short.h>
 #include "vector_table.h"
-#include "macro_priv.inc"
 
 _ASM_FILE_PROLOGUE
 
+/*
+ * Save volatile registers, x30, SPSR_EL1 and ELR_EL1
+ *
+ * Save the volatile registers and x30 on the process stack. This is
+ * needed if the thread is switched out because they can be clobbered by the
+ * ISR and/or context switch.
+ */
+
+.macro z_arm64_enter_exc xreg0, xreg1
+
+	/* Switch to SP_EL0 */
+	msr	spsel, #0
+
+	/*
+	 * Two things can happen to the remaining registers:
+	 *
+	 * - No context-switch: in this case x19-x28 are callee-saved register
+	 *   so we can be sure they are not going to be clobbered by ISR.
```

Comment:
As an aside: in my experience this is a very good invariant to add a debug check for (later, not in this PR!).

Even just storing the XOR of the callee-save registers and checking it at the end is rather helpful.

*****

There are way too many things that can result in callee-save registers being clobbered, and the resulting debugs are... well, I find them fun, but I'm pretty sure I'm the outlier there.